packages/client/lib/service/fullethereumservice.ts

```diff
@@ -94,6 +95,9 @@ export class FullEthereumService extends EthereumService {
       peer.eth!.send('BlockBodies', { reqId, bodies })
     } else if (message.name === 'NewBlockHashes') {
       this.synchronizer.handleNewBlockHashes(message.data)
+    } else if (message.name === 'NewBlock') {
+      const block = Block.fromValuesArray(message.data[0], { common: this.config.execCommon })
```

Comment:
The exec common is generally (or maybe: in the most cases) not a good approximation for instantiating chain objects, since the exec common state is often completely unpredictable since block execution often lags behind chain updates and the HF might be behind.

In this case chain common is also not a fitting choice though since the chain sync can be behind the tip of the chain. I had a similar problem when along the recent [sendRawTransaction](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/client/lib/rpc/modules/eth.ts#L469) RPC method updates, I guess you can take the pattern from there with a copied common and the HF derived from the latest `syncTargetHeight`:

```typescript
const common = this.client.config.chainCommon.copy()
common.setHardforkByBlockNumber(this.service.synchronizer.syncTargetHeight)
```

Reply:
Ah, thanks for the insight.  I've adjusted to an equivalent instance for this use case.